Main.hs

```diff
@@ -121,14 +121,14 @@ readMoves = do line <- getLine
                if line == "."
                   then return []
                   else do rest <- readMoves
-                          return (validateAndConvertMove line : rest) where
+                          return ((convertMove.validate) line : rest) where
                             err :: a
-                            err = error "invalid string for move"
+                            err = error "Unknown move"
 
-                            validateAndConvertMove :: String -> Move
-                            validateAndConvertMove [c1]       = if (c1 == 'd' || c1 == 'D') then Draw else err
-                            validateAndConvertMove [c1,c2,c3] = if (c1 == 'r' || c1 == 'R') then Discard (convertCard c2 c3) else err
-                            validateAndConvertMove _          = err
+                            validate :: String -> (Char, Char, Char)
+                            validate [c1]       = if (c1 == 'd' || c1 == 'D') then (c1, 'x', 'x') else err
```

Comment:
We already check the first character for 'Unrecognized Move' in `convertMove` function. I think this function should only check for the number of characters that are entered. So simply:
````
validate :: String -> (Char, Char, Char)
validate [c1]       = (c1, 'x', 'x')
validate [c1,c2,c3] = (c1, c2, c3)
validate _          = 'Invalid string'
````